### PR TITLE
Feature/boom

### PR DESCRIPTION
**Pull request by richardiskra**
cleaned up code and added ngDocs

### CODE REVIEW

File: src/app/routes/boom/boom.component.ts [hunk 1]
```diff
@@ -7,9 +7,70 @@ import { Component, OnInit } from '@angular/core';
...
+  public barChartType: string = 'bar';
+  public barChartLegend: boolean = true;
+
+  public barChartData: any[] = [
+    { data: this.completed, label: 'Completed' },
+    { data: this.missed, label: 'Missed' }
+  ];
+
+
+//ponut chart
```
Comment (danielsidhu) [inline src/app/routes/boom/boom.component.ts]: ponut needs to change to Donut*

File: src/app/routes/boom/boom.component.ts [hunk 2]
```diff
@@ -7,9 +7,70 @@ import { Component, OnInit } from '@angular/core';
...
   }
 
+  ngOnInit() {
+    var completed = 0;
+    var missed = 0;
+    for(var i = 0; i < this.data.curriculum.length; i++){
+      //get week titles for the axis lables
+      this.weeks.push(this.data.curriculum[i].week);
+
+      //get the completed and missed takss and count how many there are each week
```
Comment (danielsidhu) [inline src/app/routes/boom/boom.component.ts]: missed tasks*

File: src/app/routes/boom/boom.component.ts [hunk 3]
```diff
@@ -53,17 +60,17 @@ export class BoomComponent implements OnInit {
...
 
-      //get the completed and missed takss and count how many there are each week
-      for(var j = 0; j < this.data.curriculum[i].tasks.length; j++){
-        if(this.data.curriculum[i].tasks[j].status == 1){
+      // get the completed and missed takss and count how many there are each week
+      for (let j = 0; j < this.data.curriculum[i].tasks.length; j++) {
+        if (this.data.curriculum[i].tasks[j].status === 1) {
           ++completed;
-        } else if (this.data.curriculum[i].tasks[j].status == 2){
+        } else if (this.data.curriculum[i].tasks[j].status === 2) {
```
Comment (danielsidhu) [inline src/app/routes/boom/boom.component.ts:76]: loved the changes to the datatypes by reducing the redundancy.

**Comment by danielsidhu** [comment]
If I could get confirmation on whether or not you're ready to push to dev or you are still working still.

**Comment by richardiskra** [comment]
Yes, It is ready to push

**Review by danielsidhu — APPROVED** [review]
Changes were handled and branch is ready to merge

**Review by humphreyminottjr — APPROVED** [review]
Clean code

**Pull request merged**